src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+  (aaa && bbb)
+end
+active class Tiling_box
+  var boxes : [Box]
+  var futures : [Fut[bool]]
+  var futs_left : int
+  def init(agents : [Agent], n : int) : unit
+    match find_extreme(agents) with
+      case (xmax, ymax, xmin, ymin) =>
+        val dx = (xmax - xmin) / n
+        this.boxes = new [Box]( n * n )
+        this.futures = new [Fut[bool]](|this.boxes|)
+        if(n>1) then
+            var x = xmin
+            var y = ymin
+            var i = 0
+            var win = true
+            val dy = (ymax - ymin) / n
+            for xindex <- [0..n - 2] do
+              for yindex <- [0..n - 2] do
+                this.boxes(i) = new Box((x + dx - 1, y + dy - 1), (x, y))
+                if yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                end
+                if xindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                end
+                if xindex != 0 && yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+                  ()
+                end
+                i = i + 1
+                y = y + dy
+              end
+              this.boxes(i) = new Box((x + dx - 1, ymax), (x, y))
+              if xindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+              i = i + 1
+              x = x + dx
+              y = ymin
+            end
+            for yindex <- [0..n - 2] do
+              (this.boxes)(i) = new Box((xmax, y + dy - 1), (x, y))
+              if yindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+              win = win && link((this.boxes)(i), (this.boxes)(i - n))
+              i = i + 1
+              y = y + dy
+            end
+            (this.boxes)(i) = new Box((xmax, ymax), (x, y))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n))
+            win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+        else
+            this.boxes(0) = new Box((xmax,ymax), (xmin,ymin))
+        end
+        for a <- agents do
+          for b <- this.boxes do
+            if get(b!add(a)) then
+                break
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def move(i : int) : unit
+    for id <- [0..|this.boxes| - 1] do
+      (this.futures)(id) = ((this.boxes)(id)) ! move()
+    end
+    for f <- this.futures do
+      get(f)
```

Comment:
I wonder whether this can be implemented more efficiently (this is in part a note to @kikofernandez etc). For example, if all the futures are chained together and wrapped with a single `get`, then that would involve only a single `get`, which is the most expensive operation here (except when a future is fulfilled already). With three futures, the code would look like:
`get (f1 ~~> (fun (x : t) f2 ~~> (fun (x : t) f3 ~~> fun (x : t) ())))` (plus tweaking to get types etc right).
This is probably far from optimal, given that creating a future is also expensive and this would create 3 more.